### PR TITLE
Refactor architectures into library

### PR DESCRIPTION
**Pull request by EthanMarx**
Refactor architectures into library so that it can be accessed by both train and online projects

- [x] Quick end to end sanity check run of training

### CODE REVIEW

**Comment by EthanMarx** [comment]
@wbenoit26 This should be good to go - once this is in, will take a look at incorporating this into the online deployment / building a more robust CLI for that.

File: libs/priors/priors/priors.py [hunk 1]
```diff
@@ -15,7 +15,8 @@
     Uniform,
 )
 from bilby.gw.prior import UniformComovingVolume, UniformSourceFrame
-from priors.utils import (
+
+from aframe.priors.utils import (
```
Comment (wbenoit26) [inline libs/priors/priors/priors.py:19]: Does this change work? There's no longer an intermediate `aframe` directory in any of the libs.

File: projects/infer/infer/data.py [hunk 2]
```diff
@@ -5,10 +5,11 @@
 
 import h5py
 import numpy as np
-from ledger.events import EventSet, RecoveredInjectionSet
-from ledger.injections import LigoResponseSet
 from ratelimiter import RateLimiter
 
+from aframe.ledger.events import EventSet, RecoveredInjectionSet
```
Comment (wbenoit26) [inline projects/infer/infer/data.py:10]: Same question here.

File: projects/infer/infer/postprocess.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import numpy as np
-from ledger.events import EventSet
+
+from aframe.ledger.events import EventSet
```
Comment (wbenoit26) [inline projects/infer/infer/postprocess.py:3]: And here

File: projects/infer/infer/utils.py [hunk 4]
```diff
@@ -6,8 +6,8 @@
 
 import h5py
 import numpy as np
-from ledger.events import EventSet, RecoveredInjectionSet
 
+from aframe.ledger.events import EventSet, RecoveredInjectionSet
```
Comment (wbenoit26) [inline projects/infer/infer/utils.py:10]: And here

File: projects/train/result.json [hunk 5]
```diff
@@ -0,0 +1,172 @@
+{"train_loss": 0.5874887704849243, "train_loss_step": 0.5704179406166077, "valid_auroc": 0.5017630457878113, "valid_auroc_epoch": 0.5017630457878113, "train_loss_epoch": 0.5874887704849243, "epoch": 0, "step": 1804, "timestamp": 1708137659, "checkpoint_dir_name": "checkpoint_000000", "should_checkpoint": true, "done": false, "training_iteration": 1, "trial_id": "c2fa3_00116", "date": "2024-02-17_02-41-20", "time_this_iter_s": 1735.80943441391, "time_total_s": 1735.80943441391, "pid": 5319, "hostname": "asha-debug-worker-69564f4b4-hpd85", "node_ip": "10.244.89.145", "config": {"train_loop_config": {"model.learning_rate": 0.0032769218860547156, "model.pct_lr_ramp": 0.08234146755742597, "data.swap_prob": 0.11408490386772788, "data.mute_prob": 0.051934592003817066, "data.waveform_prob": 0.20777137198975693}}, "time_since_restore": 1735.80943441391, "iterations_since_restore": 1}
```
Comment (wbenoit26) [inline projects/train/result.json:1]: Looks like this was added by accident

**Review by wbenoit26 — CHANGES_REQUESTED** [review]

**Review by wbenoit26 — APPROVED** [review]

**Pull request merged**